### ISSUE TITLE
Request learning curve of QAPLIB Experiments

### ISSUE THREAD

**Issue by Bincci** [Thinklab-SJTU/ThinkMatch#36]
Hi! I am very interested in your paper _[Runzhong Wang, Junchi Yan, Xiaokang Yang. "Neural Graph Matching Network: Learning Lawler's Quadratic Assignment Problem with Extension to Hypergraph and Multiple-graph Matching." TPAMI 2021]_, and hope to reproduce the QAPLIB results showed in the paper. I followed the instructions to run the QAPLIB experiments, however, I cannot get the proper results. I looked at the learning curve in tensorboard and got something like this:

<img width="275" alt="image" src="https://user-images.githubusercontent.com/95868223/145420150-e485a9b4-5d8f-4a73-a2ba-9620dd1c482a.png">

, which basically means the neural network doesn't learn anything.

I didn't change any hyper-parameters in the config file, so I am really confused by the results. It would be nice if you can point me out the reason and show me the proper learning curve of the QAPLIB task. Thanks a lot!

**Comment by rogerwwww** [comment]
Hi there, many thanks for your interest in our paper.

The loss curve is noisy because it represents the (approximated) objective functions of different problems, and these different problems have different structures thus the loss values are always jumping. Let's say, the objective value of problem 1 may be around 200 and the objective value of problem 2 may be around 300. I recommend referring to the test results and see if you can reproduce our result. 

Besides, as we stated in our paper, you have to train and test on the same class of problems, e.g. both training and testing on the ``bur`` problems, because different types of problems have different structures, and it seems non-trivial to generalize between certain types of problems.

If you have any further questions, please feel free to post any new issues.

**Comment by Bincci** [comment]
Hi! Thank you very much for your quick reply. Actually, the learning curve I posted is generated by training and testing on the same class of problems, which is exactly the "bur" problem. I understand that the loss curve is "noisy" because although the neural network is trained on the same class of problems, the objective value could vary a lot among different instances. But in general, from my understanding, the loss curve should show a trend of converging so that we can say the neural network training is effective. Otherwise, the neural network training is meaningless. 

If this is the case, I would say the good performance of the proposed algorithm comes from some other aspects, e.g. the post-processing, instead of the neural network architecture. In other words, we can get a similar performance without training the neural network. I can do some experiments to test it, but before that, it would be great to know your thoughts and ideas.

**Comment by Bincci** [comment]
> Hi, I am sorry for the late reply and I might have overlooked the recent post.
> 
> I agree that aspects like post-processing do contribute to the performance. Just a personal note here: according to my experience, for the `bur` instances, the trained NGM model will perform better than the non-trained NGM model, though the loss curves may be as noisy as you have mentioned.

I reproduced your experiments. I am pretty sure that the non-trained NGM model performs similarly to the trained NGM model, and the results of non-trained NGM model are similar to the results you posted in the paper. TBH, the proposed model is useless, at least for QAPLIB problems.

If you have any results showing that the trained NGM performs better than non-trained NGM on QAPLIB instances, please post here.
